lib/router/router.js

```diff
@@ -15,6 +19,10 @@ export default class Router extends EventEmitter {
 
     // set up the component cache (by route keys)
     this.components = { [this.route]: { Component, err } }
+    // contain a map of response of prefetched routes
+    this.prefetchedRoutes = {}
+    this.prefetchingLockManager = new LockManager(2)
+    this.prefetchingRoutes = {}
```

Comment:
Let me give more about these states:

* prefetchedRoutes - Store already prefetched routes
* prefetchingRoutes - Keep track of currently prefetching routes. So, when we asked again to prefetch the same route, it won't happen.
* prefetchingLockManager - Make sure, we only fetch two routes at once.